### PR TITLE
DFR-2943 d11 served blank

### PR DESCRIPTION
**Pull request by al-hmcts**
### Jira link (if applicable)
[DFR-2943](https://tools.hmcts.net/jira/browse/DFR-2943)

Merge pending demo test

### Change description ###
Flatten PDFs when downloaded from doc store to be sent to bulkprint

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/DocumentConversionService.java [hunk 1]
```diff
@@ -57,6 +60,23 @@ public String getConvertedFilename(String filename) {
...
+        try {
+            PDDocument doc = PDDocument.load(document);
+            PDAcroForm acroForm = doc.getDocumentCatalog().getAcroForm();
+
+            acroForm.flatten();
+            doc.save(bos);
+            doc.close();
+            return bos.toByteArray();
+        } catch (IOException e) {
+            log.error("Unable to flatten document");
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/DocumentConversionService.java:75]: `log.error("Unable to flatten document", e);`

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentgenerator/DocumentConversionServiceTest.java [hunk 2]
```diff
@@ -91,4 +107,12 @@ public void getConvertedFilename() {
...
         assertThat(documentConversionService.getConvertedFilename("word.docx"), is("word.pdf"));
     }
+
+    private byte[] loadResource(String testPdf) throws IOException {
+
+        try (InputStream resourceAsStream = getClass().getResourceAsStream(testPdf)) {
+            assert resourceAsStream != null;
+            return resourceAsStream.readAllBytes();
+        }
+    }
```
Comment (ptrelease) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentgenerator/DocumentConversionServiceTest.java:117]: Looks good.  What happens if the flatten function gets a flat file as the param?  Looks like something easy to check by sending the flat binary into the test, and see if the method just returns it.

Comment (al-hmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentgenerator/DocumentConversionServiceTest.java:117]: @ptrelease have attempted to unit test but when flattening an additional byte of information is added and modified by PDFBox for tracking/flagging, the resulting file of flattening an already flattened PDF is an unchanged PDF that is flat. However, there is an additional byte in the result PDF that is used by PDFBox. 

The PDFBox Library also has a unit tests of there own covering this check in pdacroformflattentest.java, where the PDF before and after flattening are rendered and content is compared.

Comment (ptrelease) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentgenerator/DocumentConversionServiceTest.java:129]: Exception test looks great.  And though a test wasn't viable, al has added a comment on ln65 that answers my question

**Review by ptrelease — APPROVED** [review]

**Review by hamzahtahirhmcts — APPROVED** [review]

**Review by so99y — APPROVED** [review]
Reviewed [35e2baf](https://github.com/hmcts/finrem-case-orchestration-service/pull/1754/commits/35e2bafd6d3995ef453b03bd681ca53a79c6e26d). It makes sense to me.

File: infrastructure/state.tf [hunk 3]
```diff
@@ -2,12 +2,12 @@ terraform {
...
-      }
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.109.0"
     }
+    random = {
+      source = "hashicorp/random"
+    }
+  }
 }
```
Comment (so99y) [inline infrastructure/state.tf:13]: a new line?

File: infrastructure/variables.tf [hunk 4]
```diff
@@ -3,5 +3,5 @@ variable "product" {}
 variable "env" {}
 
 variable "common_tags" {
-    type = map(string)
+  type = map(string)
 }
```
Comment (so99y) [inline infrastructure/variables.tf:7]: a new line?

File: infrastructure/state.tf (hunk 3 above)
Comment (al-hmcts) [reply infrastructure/state.tf:13]: Autogenerated file added this, don't think this needs changing?

File: infrastructure/variables.tf (hunk 4 above)
Comment (al-hmcts) [reply infrastructure/variables.tf:7]: Autogenerated file added this, don't think this needs changing?

**Review by ptrelease — APPROVED** [review]

**Pull request merged**